tests/unit/igraph_widest_paths.c

```diff
@@ -0,0 +1,578 @@
+    igraph_vs_seq(&to, 0, n-1);
+
+    init_vertices_and_edges(n, &vertices, &edges, &predecessors, &inbound_edges, &vertices2, &edges2);
+    run_widest_paths(&g, &w, &res1, &res2, &from, &to, IGRAPH_OUT);
+    run_get_widest_paths(&g, &w, /* source */ 0, /* destination */ 4, &to, IGRAPH_OUT,
+                            &vertices, &edges, &predecessors, &inbound_edges, &vertices2, &edges2);
+
+    print_results(n, &res1, &res2, &vertices, &edges, &predecessors, &inbound_edges, &vertices2, &edges2);
+
+    destroy_all(&g, &w, &res1, &res2, &from, &to, &vertices, &edges, &predecessors, &inbound_edges,
+                &vertices2, &edges2);
+
+    /* ==================================================================== */
+    /* 13. 7 Node Wikipedia Graph                                           */
+    printf("\n=== 13. Testing 7 Node Wikipedia Graph ===\n");
+
+    n = 7;
+    m = 11;
+    igraph_small(&g, n, IGRAPH_UNDIRECTED,
+                 0, 1, 0, 2, 1, 2, 1, 3, 2, 4, 2, 5,
+                 3, 4, 3, 6, 4, 5, 4, 6, 5, 6,
+                 -1);
+    igraph_vector_init_real(&w, m, 15.0, 53.0, 40.0, 46.0, 31.0,
+                            17.0, 3.0, 11.0, 29.0, 8.0, 40.0);
+
+    igraph_matrix_init(&res1, 1, n);
+    igraph_matrix_init(&res2, 1, n);
+    igraph_vs_1(&from, 3);
+    igraph_vs_seq(&to, 0, n-1);
+
+    init_vertices_and_edges(n, &vertices, &edges, &predecessors, &inbound_edges, &vertices2, &edges2);
+
+    run_widest_paths(&g, &w, &res1, &res2, &from, &to, IGRAPH_OUT);
+    run_get_widest_paths(&g, &w, /* source */ 3, /* destination */ 6, &to, IGRAPH_OUT,
+                            &vertices, &edges, &predecessors, &inbound_edges, &vertices2, &edges2);
+
+
+    print_results(n, &res1, &res2, &vertices, &edges, &predecessors, &inbound_edges, &vertices2, &edges2);
+    IGRAPH_ASSERT(MATRIX(res1, 0, 6) == 29);
+
+    destroy_all(&g, &w, &res1, &res2, &from, &to, &vertices, &edges, &predecessors, &inbound_edges,
+                &vertices2, &edges2);
+
+
+    /* ==================================================================== */
+    /* 14. Negative Widths                                                  */
+    printf("\n=== 14. Testing Negative Weights ===\n");
+
+    n = 4;
+    m = 4;
+    igraph_small(&g, n, IGRAPH_DIRECTED, 0, 1, 1, 2, 0, 3, 3, 2, -1);
+    igraph_vector_init_real(&w, m, 2.0, -3.0, -5.0, 6.0);
+
+    igraph_matrix_init(&res1, 1, n);
+    igraph_matrix_init(&res2, 1, n);
+    igraph_vs_1(&from, 0);
+    igraph_vs_seq(&to, 0, n-1);
+
+    init_vertices_and_edges(n, &vertices, &edges, &predecessors, &inbound_edges, &vertices2, &edges2);
+
+    run_widest_paths(&g, &w, &res1, &res2, &from, &to, IGRAPH_OUT);
+    run_get_widest_paths(&g, &w, /* source */ 0, /* destination */ 2, &to, IGRAPH_OUT,
+                            &vertices, &edges, &predecessors, &inbound_edges, &vertices2, &edges2);
+
+    IGRAPH_ASSERT(MATRIX(res1, 0, 2) == -3);
+    check_and_print_matrices(&res1, &res2);
```

Comment:
Why don't you do a `print_results` here?

Reply:
Great question :) This has been fixed.